### PR TITLE
Sh 237 - user self claimed external-id report

### PR DESCRIPTION
**Pull request by Hari-stackroute**
It has code changes for generating user-self submitted external id detail report.
reports are generated against channel in the admin-user-reports/declared_user_detail/{{channel.csv}}
Email and phone values are decrypting to normal-text(readable-format) while generating the report.

### CODE REVIEW

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala [hunk 1]
```diff
@@ -49,14 +53,75 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
...
+        DecryptUtil.initialise();
         execute(jobConfig)
         closeSparkSession()
     }
-
+    
     private def execute(config: JobConfig)(implicit sparkSession: SparkSession, fc: FrameworkContext) = {
 
-        generateReport();
+        /*generateReport();*/
```
Comment (indrajra) [inline data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala:63]: make this live again!

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala [hunk 2]
```diff
@@ -49,14 +53,75 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
...
+        import sparkSession.implicits._
+    
+        val container = AppConf.getConfig("cloud.container.reports")
+        val objectKey = AppConf.getConfig("admin.metrics.cloud.objectKey")
+        val storageConfig = getStorageConfig(container, objectKey);
+        val usrExternalIdentityEncoder = Encoders.product[UsrExternalIdentity].schema
+        val userExternalDataDF = loadData(sparkSession, Map("table" -> "usr_external_identity", "keyspace" -> sunbirdKeyspace), Some(usrExternalIdentityEncoder)).
+            filter(col(colName = "idtype").isin("declared-email", "declared-phone", "declared-school-name", "declared-school-udise-code", "declared-ext-id", "declared-state", "declared-district")).as[UsrExternalIdentity]
+        val locationDF = locationData();
+        val userExternalOriginalDataDF = userExternalDataDF.groupBy("userid", "provider").pivot("idtype").agg(first("originalexternalid").alias("originalexternalid"))
```
Comment (indrajra) [inline data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala:80]: pls put a comment why origExternalId is being looked up here.

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala [hunk 3]
```diff
@@ -49,14 +53,75 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
...
+        
+        val userExternalStateDF = userExternalOriginalDataDF.join(locationDF, userExternalOriginalDataDF.col("declared-state") === locationDF.col("locid") && locationDF.col("loctype") === "state", "left_outer").
+                select(userExternalOriginalDataDF.col("*"), locationDF("locname").as("state"))
+        val userExternalLocationDF = userExternalStateDF.join(locationDF, userExternalOriginalDataDF.col("declared-district") === locationDF.col("locid") &&locationDF.col("loctype") === "district", "left_outer").
```
Comment (indrajra) [inline data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala:84]: should this be named userExternalDistrictDF ?

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala (hunk 1 above)
Comment (Hari-stackroute) [reply data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala]: done.

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala (hunk 2 above)
Comment (Hari-stackroute) [reply data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala]: comments are added across the code

**Pull request merged**